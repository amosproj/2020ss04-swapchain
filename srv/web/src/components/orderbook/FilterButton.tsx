import React, { useState } from "react"

type Props = {
    onClick: (s: string) => void
    label: string
}

export const FilterButton = (props: Props) => {
    const { onClick, label } = props

    const [active, setActive] = useState(false)

    const toggle = () => {
        setActive(!active)
        onClick(label)

    }



    return (
        <button
            onClick={toggle}
            className={`focus:outline-none px-2 py-1 mx-2 text-xs font-semibold  uppercase  rounded hover:bg-gray-300 hover:text-gray-900 text-gray-800 ${active ? "bg-green-200" : "bg-gray-100"}`}>
            {label}
        </button>
    )








}