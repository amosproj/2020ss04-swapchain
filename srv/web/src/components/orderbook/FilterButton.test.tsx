import React from 'react';
import {FilterButton} from './FilterButton';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer
    .create(<FilterButton key="1" label="label" onClick={()=>{}}></FilterButton>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});