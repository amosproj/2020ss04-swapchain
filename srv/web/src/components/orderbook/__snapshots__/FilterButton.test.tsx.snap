// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  className="focus:outline-none px-2 py-1 mx-2 text-xs font-semibold  uppercase  rounded hover:bg-gray-300 hover:text-gray-900 text-gray-800 bg-gray-100"
  onClick={[Function]}
>
  label
</button>
`;
